python/src/aac/plugins/gen_plugin/gen_plugin_impl.py

```diff
@@ -171,7 +172,7 @@ def _prepare_and_generate_plugin_files(
     template_output_directories = _get_template_output_directories(
         plugin_type, architecture_file_path, plugin_implementation_name
     )
-
+    _move_architecture_file_to_plugin_root(architecture_file_path, plugin_implementation_name, output_directory)
```

Comment:
This should probably go inside the `_generate_plugin_files_to_directory` function after `write_generated_templates_to_file`.

Reply:
FYSA: `write_generated_templates_to_file` eventually calls `write_file` which will [create directories if they don't exist](https://github.com/jondavid-black/AaC/blob/974578f1102e9f0edca55bb2caef685db93d2059/python/src/aac/io/writer.py#L24-L25) and, until that's called the plugin directory may not exist.

Reply:
I think this should be good now. Let me know if otherwise.